fmf/utils.py

```diff
@@ -500,6 +501,57 @@ def enabled(self):
         return self._mode == COLOR_ON
 
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# checkout_remote
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+def checkout_remote(remote_ref):
+    """
+    Checkouts remote_ref to the cache dir and returns local directory.
+
+    remote_ref: Dictionary following Remote reference proposal (#52 issue).
+                See base.Tree.node for description
+                Key 'name' is ignored.
+    """
+
+    default_for_rev = 'master'
+
+    destination = os.path.join(
+        os.path.join(os.path.expanduser(os.environ.get('XDG_CACHE_HOME', '~/.cache')), 'fmf'),
+        remote_ref['url'].replace('/', '_'),
+        remote_ref.get('ref', default_for_rev).replace('/', '_'),
```

Comment:
Is it necessary to create a separate directory for each `ref`? Would it be ok to have just a single directory for each `url` under the cache and just use `git checkout` for switching refs? Especially for larger repositories this could save some space.

Reply:
Or is this needed to prevent possible race conditions between multiple processes?

Reply:
I was trying to avoid git checkout every time. But doing it every time might be better

Reply:
I see. Ok, checkout always seems to me a bit better to save resources with fetching repos.